lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -3795,6 +3795,11 @@ static void recordConformanceDependency(DeclContext *DC,
       Conformance->getDeclContext()->getParentModule())
     return;
 
+  // Register a potential member dependency edge to the adoptee.
+  // This ensures the file containing the conformance rebuilds if the adoptee
+  // makes a change to its members. This is especially important if the
+  // conformance is defined in an extension context: this edge ensures the
+  // extension's file rebuilds when the extended type changes.
```

Comment:
Had to read this twice. How could it be clearer? Hmmm... How about something like this:
"If a type (or protocol?) or extension declaration states that it conforms to some protocol, that declaration must be recompiled if the conformed-to protocol changes."

Reply:
The dependency edge doesn't make reference to changes in the protocol, only in the adopting type. Those potential member edges are added by `TypeChecker::conformsToProtocol`.

Reply:
This function is a little confusingly named. It's not registering a dependency on a protocol, it's ensuring that an `extension` rebuilds if its extended type changes at all because we have no other way of expressing that constraint today.